### ISSUE TITLE
Random `DRb::DRbConnError` Errors

### ISSUE THREAD

**Issue by metaskills** [customink/lambda_punch#2]
Under load and using CloudWatch queries I was able to find random errors like this.

```
loop -> api.rb:20 -> api.rb:60 -> worker.rb:56
DRb::DRbConnError
connection closed
```

I was able to count these using this CloudWatch Insights query.

```
fields @message
| filter @message like 'DRb::DRbConnError'
| sort @timestamp desc
```

**Comment by metaskills** [comment]
While debugging these I was able to see how the function never got the rb-inotify event and hence waited for 60s the functions timeout. See below.

```
>> Time.at(1624967717774/1000.0)  => 2021-06-29 07:55:17 3246391/4194304 -0400
>> Time.at(1624967777782/1000.0)  => 2021-06-29 07:56:17 1639973/2097152 -0400

| 1624967717774 | START RequestId: 2162d887-7499-4f83-9aa8-7d2cfc4dcdab Version: 54                                                                                   |
| 1624967777782 | [DEBUG] 2162d887-7499-4f83-9aa8-7d2cfc4dcdab                                                                                                        |
| 1624967777919 | /var/lang/lib/ruby/2.7.0/drb/drb.rb:743:in `rescue in block in open'                                                                                |
| 1624967777919 | :                                                                                                                                                   |
| 1624967777919 | druby://127.0.0.1:9030 - #<Errno::ECONNREFUSED: Connection refused - connect(2) for "127.0.0.1" port 9030>                                          |
| 1624967777919 |  (                                                                                                                                                  |
| 1624967777919 | DRb::DRbConnError                                                                                                                                   |
| 1624967777919 | )                                                                                                                                                   |
| 1624967777919 |                                                                                                                                                     |
| 1624967777919 |  from /var/lang/lib/ruby/2.7.0/drb/drb.rb:737:in `block in open'                                                                                    |
| 1624967777919 |  from /var/lang/lib/ruby/2.7.0/drb/drb.rb:736:in `each'                                                                                             |
| 1624967777919 |  from /var/lang/lib/ruby/2.7.0/drb/drb.rb:736:in `open'                                                                                             |
| 1624967777919 |  from /var/lang/lib/ruby/2.7.0/drb/drb.rb:1318:in `initialize'                                                                                      |
| 1624967777919 |  from /var/lang/lib/ruby/2.7.0/drb/drb.rb:1301:in `new'                                                                                             |
| 1624967777919 |  from /var/lang/lib/ruby/2.7.0/drb/drb.rb:1301:in `open'                                                                                            |
| 1624967777919 |  from /var/lang/lib/ruby/2.7.0/drb/drb.rb:1142:in `block in method_missing'                                                                         |
| 1624967777919 |  from /var/lang/lib/ruby/2.7.0/drb/drb.rb:1161:in `with_friend'                                                                                     |
| 1624967777919 |  from /var/lang/lib/ruby/2.7.0/drb/drb.rb:1141:in `method_missing'                                                                                  |
| 1624967777919 |  from /var/task/vendor/bundle/ruby/2.7.0/gems/lambda_punch-0.0.7/lib/lambda_punch/worker.rb:36:in `rescue in call_queue'                            |
| 1624967777919 |  from /var/task/vendor/bundle/ruby/2.7.0/gems/lambda_punch-0.0.7/lib/lambda_punch/worker.rb:31:in `call_queue'                                      |
| 1624967777919 |  from /var/task/vendor/bundle/ruby/2.7.0/gems/lambda_punch-0.0.7/lib/lambda_punch/worker.rb:72:in `call'                                            |
| 1624967777919 |  from /var/task/vendor/bundle/ruby/2.7.0/gems/lambda_punch-0.0.7/lib/lambda_punch/worker.rb:26:in `call'                                            |
| 1624967777919 |  from /var/task/vendor/bundle/ruby/2.7.0/gems/lambda_punch-0.0.7/lib/lambda_punch/api.rb:67:in `invoke'                                             |
| 1624967777919 |  from /var/task/vendor/bundle/ruby/2.7.0/gems/lambda_punch-0.0.7/lib/lambda_punch/api.rb:45:in `loop'                                               |
| 1624967777920 |  from /var/task/vendor/bundle/ruby/2.7.0/gems/lambda_punch-0.0.7/lib/lambda_punch/api.rb:20:in `loop'                                               |
| 1624967777920 |  from /var/task/vendor/bundle/ruby/2.7.0/gems/lambda_punch-0.0.7/lib/lambda_punch.rb:29:in `loop'                                                   |
| 1624967777920 |  from /opt/extensions/lambdapunch:11:in `<main>'                                                                                                    |
| 1624967777920 | /var/lang/lib/ruby/2.7.0/drb/drb.rb:839:in `initialize'                                                                                             |
| 1624967777920 | :                                                                                                                                                   |
| 1624967777920 | Connection refused - connect(2) for "127.0.0.1" port 9030                                                                                           |
| 1624967777920 |  (                                                                                                                                                  |
| 1624967777920 | Errno::ECONNREFUSED                                                                                                                                 |
| 1624967777920 | ) 
```

**Comment by metaskills** [comment]
Before I noticed that pattern above I tried to re-create the DRb connection. This was not needed but sharing:

* https://github.com/customink/lambda_punch/commit/519288152bcf8d288df0d703667c1d182e18ce7a
* https://github.com/customink/lambda_punch/commit/ab911c7b4054e6fb78f348e2db8e108c0816d8c3

The real fix was making sure we check at the very last minute in the Worker#call method that the notify event did not happen already. That was done here.

https://github.com/customink/lambda_punch/commit/ccbe358a8d6c14cf7575fabca2b47882ea0a5e4c

Afterward I ran upwards of 200,000 invocations at various durations and never had a single failure or lost job.
